/**
  * All the GUI settings.
  */
  
import "common.proto";
  
package Protos.GUI;

message Settings {
   optional string core_address = 1 [default = "localhost"];
   optional uint32 core_port = 2 [default = 59485];   
   optional Common.Hash password = 3; // Hashed with the salt given by the core. See the 'Protos.GUI.AskForAuthentication' message.
   
   optional Common.Language language = 20;
      
   optional uint32 max_chat_message_displayed = 4 [default = 1000];
   optional uint32 max_log_message_displayed = 5 [default = 500];
   optional uint32 search_time = 7 [default = 3000]; // [ms].
   optional uint32 socket_timeout = 8 [default = 7000]; // [ms].
   
   optional bytes windows_state = 9;
   optional uint32 main_window_width = 10 [default = 900];   
   optional uint32 main_window_height = 11 [default = 450];
   optional bool main_window_maximized = 17 [default = false];
   
   repeated uint32 search_column_size = 19; // [Entry name, Folder name, Relvance, Peer, Size].
   
   enum Window {
      WIN_SETTINGS = 1;
      WIN_CHAT = 2;
      WIN_DOWNLOAD = 3;
      WIN_UPLOAD = 4;
   }   
   repeated Window windowOrder = 18;   
   
   enum DownloadView {
      TREE_VIEW = 1;
      LIST_VIEW = 2;
   }
   optional DownloadView download_view = 21 [default = TREE_VIEW];
   
   enum PeerSortType {
      BY_NICK = 1;
      BY_SHARING_AMOUNT = 2;
   }
   optional PeerSortType peer_sort_type = 22 [default = BY_SHARING_AMOUNT];
   
   message HighlightedPeers {
      message Peer {
         required uint32 color = 1; // RGB.
         required Common.Hash id = 2;
      }
      repeated Peer peer = 1;
   }
   optional HighlightedPeers highlighted_peers = 23;
}
